### PR TITLE
feat: add-generic migration guides

### PR DESCRIPTION
**Pull request by mansueli**
## What kind of change does this PR introduce?

Includes generic guides for Postgres, MySQL and MSSQL migrations to database.

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by KevinBrolly** [comment]
Nice @mansueli this is 🔥

File: apps/docs/pages/guides/resources.mdx [hunk 1]
```diff
@@ -147,6 +147,24 @@ export const migrationGuides = [
...
+    title: 'Postgres',
+    icon: '/docs/img/icons/postgres-icon',
+    href: '/guides/resources/migrating-to-supabase/postgres',
+    description: 'Migrate your Postgres database to Supabase.',
+  },
+  {
+    title: 'MySQL',
+    icon: '/docs/img/icons/mysql-icon',
+    href: '/guides/resources/migrating-to-supabase/mysql',
+    description: 'Migrate your MySQLdatabase to Supabase.',
```
Comment (gregnr) [inline apps/docs/pages/guides/resources.mdx:160]:
```suggestion
    description: 'Migrate your MySQL database to Supabase.',
```

File: apps/docs/pages/guides/resources/migrating-to-supabase/amazon-rds.mdx [hunk 2]
```diff
@@ -96,8 +96,11 @@ pgloader config.load
...
 
-- If you plan on migrating a database larger than 6 GB, we recommend [contacting support](https://supabase.com/dashboard/support/new) to ensure that you'll have the proper disk size pre-provisioned. You can read more about how the disk is managed on Supabase on the [Database usage](/docs/guides/platform/database-size#disk-management).
-- We also recommend upgrading to at least a [Large instance](/docs/guides/platform/compute-add-ons) for the migration (you can downgrade later) to avoid running out of IO-Budget.
+- If you're planning to migrate a database larger than 6 GB, we recommend [upgrading to at least a Large compute add-on](/docs/guides/platform/compute-add-ons). This will ensure you have the necessary resources to handle the migration efficiently.
+
+- For databases smaller than 150 GB, you can increase the size of the disk on paid projects by navigating to [Database Settings](/dashboard/project/_/settings/database).
+
+- If you're dealing with a database larger than 150 GB, we strongly advise you to [contact our support team](/dashboard/support/new) for assistance in provisioning the required resources and ensuring a smooth migration process.
 
 </Admonition>
```
Comment (gregnr) [inline apps/docs/pages/guides/resources/migrating-to-supabase/amazon-rds.mdx:105]: PS. for duplicate snippets like this, consider saving them under `./apps/docs/components/MDX` then reuse them by importing that file as a JSX component.

File: apps/docs/pages/guides/resources/migrating-to-supabase/mssql.mdx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+<Tabs
+  scrollable
+  size="small"
+  type="underlined"
+  defaultActiveId="colab"
+  queryGroup="migrate-method"
+>
+<TabPanel id="colab" label="Migrate using Colab">
+
+1. Select the Dabase Engine from the Source database in the dropdown
```
Comment (gregnr) [inline apps/docs/pages/guides/resources/migrating-to-supabase/mssql.mdx:46]:
```suggestion
1. Select the Dabase Engine from the Source database in the dropdown.
```

File: apps/docs/pages/guides/resources/migrating-to-supabase/mssql.mdx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+1. Save your password or [reset it](https://supabase.com/dashboard/project/_/settings/database) if you forgot it.
+
+![Finding Supabase host address](/docs/img/guides/resources/migrating-to-supabase/amazon-rds/supabase_dashboard.png)
+
+## Migrate the database
+
+The fastest way to migrate your database is with the Supabase migration tool on
+[Google Colab](https://colab.research.google.com/github/mansueli/Supa-Migrate/blob/main/Amazon_RDS_to_Supabase.ipynb).
+
+Alternatively, you can use [pgloader](https://github.com/dimitri/pgloader), a flexible and powerful data migration tool that supports a wide range of source database engines, including MySQL and MS SQL, and migrates the data to a Postgres database. For databases using the Postgres engine, we recommend using the pg_dump and psql command line tools, which are included in a full PostgreSQL installation.
```
Comment (gregnr) [inline apps/docs/pages/guides/resources/migrating-to-supabase/mssql.mdx:35]:
```suggestion
Alternatively, you can use [pgloader](https://github.com/dimitri/pgloader), a flexible and powerful data migration tool that supports a wide range of source database engines, including MySQL and MS SQL, and migrates the data to a Postgres database. For databases using the Postgres engine, we recommend using the [`pg_dump`](https://www.postgresql.org/docs/current/app-pgdump.html) and psql command line tools, which are included in a full PostgreSQL installation.
```

File: apps/docs/pages/guides/resources/migrating-to-supabase/postgres.mdx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+export const meta = {
+  title: 'Migrate from Postgres to Supabase',
+  description: 'Migrate your Postgres database to Supabase.',
+}
+
+This is a guide for migrating your Postgres database to [Supabase](https://supabase.com).
+Supabase is one of the best free Postgres providers. Supabase provide all the backend features developers need to build a product: a Postgres database, authentication, instant APIs, edge functions, realtime subscriptions, and storage. Postgres is the core of Supabase—for example, you can use row-level security and there are more than 40 Postgres extensions available.
+
+This guide demonstrates how to migrate your Postgres dataase to Supabase to get the most out of Postgres while gaining access to all the features you need to build a project.
```
Comment (gregnr) [inline apps/docs/pages/guides/resources/migrating-to-supabase/postgres.mdx:11]:
```suggestion
This guide demonstrates how to migrate your Postgres database to Supabase to get the most out of Postgres while gaining access to all the features you need to build a project.
```

File: apps/docs/pages/guides/resources/migrating-to-supabase/postgres.mdx [hunk 6]
```diff
@@ -0,0 +1,95 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  title: 'Migrate from Postgres to Supabase',
+  description: 'Migrate your Postgres database to Supabase.',
+}
+
+This is a guide for migrating your Postgres database to [Supabase](https://supabase.com).
+Supabase is one of the best free Postgres providers. Supabase provide all the backend features developers need to build a product: a Postgres database, authentication, instant APIs, edge functions, realtime subscriptions, and storage. Postgres is the core of Supabase—for example, you can use row-level security and there are more than 40 Postgres extensions available.
```
Comment (gregnr) [inline apps/docs/pages/guides/resources/migrating-to-supabase/postgres.mdx:9]: Unsure about the wording around "Supabase is one of the best free Postgres providers ..." within our own docs. Perhaps we can use the same wording as the other migration guides, like "Supabase is a robust and open-source platform ..."?

File: apps/docs/pages/guides/resources/migrating-to-supabase/mysql.mdx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+1. Save your password or [reset it](https://supabase.com/dashboard/project/_/settings/database) if you forgot it.
+
+![Finding Supabase host address](/docs/img/guides/resources/migrating-to-supabase/amazon-rds/supabase_dashboard.png)
+
+## Migrate the database
+
+The fastest way to migrate your database is with the Supabase migration tool on
+[Google Colab](https://colab.research.google.com/github/mansueli/Supa-Migrate/blob/main/Amazon_RDS_to_Supabase.ipynb).
+
+Alternatively, you can use [pgloader](https://github.com/dimitri/pgloader), a flexible and powerful data migration tool that supports a wide range of source database engines, including MySQL and MS SQL, and migrates the data to a Postgres database. For databases using the Postgres engine, we recommend using the pg_dump and psql command line tools, which are included in a full PostgreSQL installation.
```
Comment (gregnr) [inline apps/docs/pages/guides/resources/migrating-to-supabase/mysql.mdx:36]: Noticing some guides have `pg_dump` and `psql` linked, and some don't. Could we keep it consistent across each guide?

File: apps/docs/pages/guides/resources/migrating-to-supabase/mssql.mdx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+```sql
+LOAD DATABASE
+    FROM mssql://USER:PASSWORD@HOST/SOURCE_DB
+    INTO postgres://postgres:password@db.xxxx.supabase.co:6543/postgres
+ALTER SCHEMA 'public' OWNER TO 'postgres';
+set wal_buffers = '64MB', max_wal_senders = 0, statement_timeout = 0, work_mem to '2GB';
+```
+
+##Run the migration with pgloader
```
Comment (gregnr) [inline apps/docs/pages/guides/resources/migrating-to-supabase/mssql.mdx:65]:
```suggestion
## Run the migration with pgloader
```

**Review by gregnr — APPROVED** [review]

**Pull request merged**